lib/matplotlib/dates.py

```diff
@@ -732,20 +733,105 @@ def __call__(self, x, pos=None):
 
 
 class rrulewrapper(object):
+    def __init__(self, freq, tzinfo=None, **kwargs):
+        kwargs['freq'] = freq
+        self._base_tzinfo = tzinfo
 
-    def __init__(self, freq, **kwargs):
-        self._construct = kwargs.copy()
-        self._construct["freq"] = freq
-        self._rrule = rrule(**self._construct)
+        self._update_rrule(**kwargs)
 
     def set(self, **kwargs):
         self._construct.update(kwargs)
+
+        self._update_rrule(**self._construct)
+
+    def _update_rrule(self, **kwargs):
+        tzinfo = self._base_tzinfo
+
+        # rrule does not play nicely with time zones - especially pytz time
+        # zones, it's best to use naive zones and attach timezones once the
+        # datetimes are returned
+        if 'dtstart' in kwargs:
+            dtstart = kwargs['dtstart']
+            if dtstart.tzinfo is not None:
```

Comment:
could do `dtstart = kwargs.get('dtstart', None)`, and therefore be able to reduce all of this by an indent.

Reply:
@WeatherGod Actually, looking at this, I don't think you *can* reduce all this by an indent, because if `dtstart` is `None`, the `dtstart.tzinfo is None` and `dtstart.replace()` calls will fail.

Reply:
actually, it'll fail for a different reason. dtstart would be `None` and it wouldn't have the `tzinfo` attribute. Don't know why I didn't see that originally.